Lib/test/test_winreg.py

```diff
@@ -318,6 +318,46 @@ def run(self):
             DeleteKey(HKEY_CURRENT_USER, test_key_name+'\\changing_value')
             DeleteKey(HKEY_CURRENT_USER, test_key_name)
 
+    def test_queryvalueex_race_condition(self):
+        # gh-142282: QueryValueEx could read garbage buffer under race
+        # condition when another thread changes the value size
+        done = False
+        error_found = None
+        values = [b'ham', b'spam']
+
+        class WriterThread(threading.Thread):
+            def run(self):
+                with CreateKey(HKEY_CURRENT_USER, test_key_name) as key:
+                    use_first = True
+                    while not done:
+                        val = values[0] if use_first else values[1]
+                        use_first = not use_first
+                        SetValueEx(key, 'test_value', 0, REG_BINARY, val)
+
+        thread = WriterThread()
+        thread.start()
+        try:
+            with CreateKey(HKEY_CURRENT_USER, test_key_name) as key:
+                for _ in range(1000):
+                    try:
+                        result, typ = QueryValueEx(key, 'test_value')
+                    except FileNotFoundError:
+                        # Value not yet created
+                        continue
```

Comment:
Several first iterations can be a blank shot. In worst case this loop can finish before the writer thread starts. I the test is repeated several times, you can see here the result of the previous test before the writer thread starts.

You can use an event. Set it in the writer after the first write and wait for it in the main thread before starting the loop. Then read will never fail.